### PR TITLE
feat: send event on checkbox change

### PR DESCRIPTION
**Pull request by jansenk**
When a user toggles the "social sharing" checkbox, sent an analytics event

This also includes a fix to an issue where we were requesting an old URL for the video features URL

### CODE REVIEW

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

File: src/editors/containers/VideoEditor/components/VideoSettingsModal/components/SocialShareWidget/constants.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const analyticsEvents = {
+  socialSharingSettingChanged: 'edx.social.video_sharing_setting.changed',
```
Comment (muselesscreator) [inline src/editors/containers/VideoEditor/components/VideoSettingsModal/components/SocialShareWidget/constants.js:2]: XD good pattern :-)

File: src/editors/data/redux/thunkActions/requests.js [hunk 2]
```diff
@@ -290,7 +290,6 @@ export const fetchVideoFeatures = ({ ...rest }) => (dispatch, getState) => {
     requestKey: RequestKeys.fetchVideoFeatures,
     promise: api.fetchVideoFeatures({
       studioEndpointUrl: selectors.app.studioEndpointUrl(getState()),
-      learningContextId: selectors.app.learningContextId(getState()),
```
Comment (muselesscreator) [inline src/editors/data/redux/thunkActions/requests.js:293]: What is the purpose of this removal?

File: src/editors/containers/VideoEditor/components/VideoSettingsModal/components/SocialShareWidget/hooks.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { useSelector } from 'react-redux';
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
+import { selectors } from '../../../../../../data/redux';
+import analyticsEvents from './constants';
+
+export const handleSocialSharingCheckboxChange = ({ updateField }) => {
```
Comment (muselesscreator) [inline src/editors/containers/VideoEditor/components/VideoSettingsModal/components/SocialShareWidget/hooks.jsx:6]: this should have a hook name (useTrackSocialSharingChange)

File: src/editors/data/redux/thunkActions/requests.js (hunk 2 above)
Comment (jansenk) [reply src/editors/data/redux/thunkActions/requests.js:293]: This is a separate bug. We are requesting the URL with the course ID at the end, but the URL was changed to be course agnostic. This request is 404ing on prod

**Review by kenclary — APPROVED** [review]
lgtm

**Pull request merged**